### PR TITLE
Convert RecentSearchSpec.java into kotlin

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes: #3005
Parent Issue: #1826 

Converted RecentSearchSpec.java into kotlin

### CODE REVIEW

**Comment by kelson42** [comment]
@gouri-panda This one as well could be easy to review/merge

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/local/entity/RecentSearchSpec.kt [hunk 1]
```diff
@@ -15,16 +15,19 @@
...
 @TableModelSpec(className = "RecentSearch", tableName = "recentSearches")
-public class RecentSearchSpec {
-  @NotNull public String searchString;
-  @NotNull String zimID;
+class RecentSearchSpec {
+  @NotNull
+  lateinit var searchString: String
+
+  @NotNull
+  lateinit var zimID: String
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/local/entity/RecentSearchSpec.kt:32]: Why it should be late initializer property?

Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/local/entity/RecentSearchSpec.kt:32]: hi @gouri-panda ,
we should make this variables null by default?

Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/local/entity/RecentSearchSpec.kt:32]: >Why it should be late initializer property?

That was a question. Do you think nullable is a good idea with `zimId`?

Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/local/entity/RecentSearchSpec.kt:32]: Hi @gouri-panda ,

we are using ```@NonNull``` annotation of ```com.yahoo.squidb.annotations.tables.constraints.NotNull``` on both ```searchString``` and ```zimId``` so on runtime these variables are treat as non null.

**Comment by kelson42** [comment]
@MohitMaliFtechiz @gouri-panda Possible to move forward with this PR. Not sure but maybe there is not much left to do?!

**Comment by kelson42** [comment]
@MohitMaliFtechiz @gouri-panda ?

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/local/entity/RecentSearchSpec.kt (hunk 1 above)
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/local/entity/RecentSearchSpec.kt:32]: hi @gouri-panda , @kelson42 
this PR is ready for re-review.

**Review by gouri-panda — APPROVED** [review]

**Pull request merged**